### PR TITLE
I2 18 items index stats

### LINKED ISSUE #20: User Story 18, Items Index Page Statistics
As any kind of user on the system
When I visit the items index page ("/items")
I see an area with statistics:
- the top 5 most popular items by quantity purchased, plus the quantity bought
- the bottom 5 least popular items, plus the quantity bought

"Popularity" is determined by total quantity of that item ordered

### PR DESCRIPTION
**Pull request by ryandbarnett**
Working but need to refactor raw SQL in model helper methods into active record

### CODE REVIEW

File: app/models/item.rb [hunk 1]
```diff
@@ -25,4 +25,14 @@ def no_orders?
     item_orders.empty?
   end
 
+  def total_bought
+    orders.sum(:quantity)
+  end
+
+  def self.most_popular_items
```
Comment (PhilipDeFraties) [inline app/models/item.rb:32]: Nice

File: app/models/item.rb [hunk 2]
```diff
@@ -25,4 +25,14 @@ def no_orders?
...
+    return joins(:item_orders)
+      .group(:id)
+      .order('SUM(item_orders.quantity) DESC')
+      .limit(5)
+  end
```
Comment (PhilipDeFraties) [inline app/models/item.rb:37]: Nice

File: spec/features/items/index_spec.rb [hunk 3]
```diff
@@ -3,111 +3,104 @@
...
+      @item_order_2 = create(:item_order, quantity: 9)
+      @item_order_3 = create(:item_order, quantity: 8)
+      @item_order_4 = create(:item_order, quantity: 7)
+      @item_order_5 = create(:item_order, quantity: 6)
+      @item_order_6 = create(:item_order, quantity: 5)
+      @item_order_7 = create(:item_order, quantity: 4)
+      @item_order_8 = create(:item_order, quantity: 3)
+      @item_order_9 = create(:item_order, quantity: 2)
+      @item_order_10 = create(:item_order, quantity: 1)
+    end
```
Comment (PhilipDeFraties) [inline spec/features/items/index_spec.rb:20]: This does look so much better

**Review by PhilipDeFraties — APPROVED** [review]
Im still working on my alternative but this is working so I'm just going to approve, nice work Ryan

File: app/views/items/index.html.erb [hunk 4]
```diff
@@ -25,3 +25,19 @@
     <% end %>
   <% end %>
 </section>
+<section id="items-statistics">
```
Comment (dhalverson) [inline app/views/items/index.html.erb:28]: NIIICE work here, looking good

**Review by dhalverson — APPROVED** [review]
Looks good, we can sync up as a group on the AR refactor.

File: spec/features/items/index_spec.rb (hunk 3 above)
Comment (ryandbarnett) [reply spec/features/items/index_spec.rb:20]: Yes I like it a lot more.

**Pull request merged**